### PR TITLE
Event Plugin

### PR DESCRIPTION
**Pull request by shazmasiddiqui**
Adds event as a new schema type

### CODE REVIEW

File: administrator/components/com_admin/sql/updates/mysql/4.2.3-2022-12-10.sql [hunk 1]
```diff
@@ -10,4 +10,8 @@ CREATE TABLE IF NOT EXISTS `#__schemaorg` (
 	`schemaForm` text,
 	`schema` text,
 	PRIMARY KEY (`id`)
-) ENGINE=INNODB DEFAULT CHARSET=utf8 AUTO_INCREMENT=1;
\ No newline at end of file
+) ENGINE=INNODB DEFAULT CHARSET=utf8;
```
Comment (bembelimen) [inline administrator/components/com_admin/sql/updates/mysql/4.2.3-2022-12-10.sql]: Please check the charset of DB tables: https://github.com/joomla/joomla-cms/blob/4.2-dev/installation/sql/mysql/base.sql#L21

File: administrator/components/com_admin/sql/updates/postgresql/4.2.3-2022-12-10.sql [hunk 2]
```diff
@@ -11,4 +11,8 @@ CREATE TABLE IF NOT EXISTS "#__schemaorg"
   "schemaForm" text,
   "schema" text,
   PRIMARY KEY ("id")
```
Comment (bembelimen) [inline administrator/components/com_admin/sql/updates/postgresql/4.2.3-2022-12-10.sql:13]: Don't forget charset (see other comment)

File: plugins/schemaorg/event/src/Extension/Event.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+     *
+     * @since   4.0.0
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'onSchemaPrepareData'                  => 'onSchemaPrepareData',
+            'onSchemaPrepareForm'                  => 'onSchemaPrepareForm',
+            'onSchemaAfterSave'                    => 'onSchemaAfterSave',
+            'onSchemaBeforeCompileHead'            => 'pushSchema',
```
Comment (bembelimen) [inline plugins/schemaorg/event/src/Extension/Event.php]: Probably too many spaces?

File: plugins/schemaorg/event/src/Extension/Event.php [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+     *  @param   $data  The form to be altered.
+     *
+     *  @return  boolean
+     */
+    public function onSchemaPrepareData(AbstractEvent $event)
+    {
+        $context = $event->getArgument('context');
+        if (!$this->isSupported($context) || !$this->isSchemaSupported($event)) {
+            return false;
+        }
```
Comment (bembelimen) [inline plugins/schemaorg/event/src/Extension/Event.php:103]: Please add empty line

File: plugins/schemaorg/event/src/Extension/Event.php [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    /**
+     *  Update existing schema form with data from database
+     *
+     *  @param   $data  The form to be altered.
+     *
+     *  @return  boolean
+     */
+    public function onSchemaPrepareData(AbstractEvent $event)
+    {
+        $context = $event->getArgument('context');
```
Comment (bembelimen) [inline plugins/schemaorg/event/src/Extension/Event.php:99]: Please add empty line

File: plugins/schemaorg/event/src/Extension/Event.php [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+     *
+     *  @return  boolean
+     */
+    public function onSchemaPrepareForm(AbstractEvent $event)
+    {
+        $form = $event->getArgument('subject');
+        $context = $form->getName();
+        if (!$this->isSupported($context)) {
+            return false;
+        }
```
Comment (bembelimen) [inline plugins/schemaorg/event/src/Extension/Event.php:124]: Please add empty line

File: plugins/schemaorg/event/src/Extension/Event.php [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        $this->addSchemaType($event);
```
Comment (bembelimen) [inline plugins/schemaorg/event/src/Extension/Event.php:126]: Please add empty line

File: plugins/schemaorg/event/src/Extension/Event.php [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        //Load the form fields
+        $form->loadFile(JPATH_PLUGINS . '/' . $this->_type . '/' . $this->_name . '/src/forms/schemaorg.xml');
```
Comment (bembelimen) [inline plugins/schemaorg/event/src/Extension/Event.php:129]: Please add empty line

File: plugins/schemaorg/event/src/Extension/Event.php [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+     *  @return  boolean
+     */
+    public function onSchemaAfterSave(AbstractEvent $event)
+    {
+        $data = $event->getArgument('data')->toArray();
+        $form = $data['schema']['schemaType'];
+
+        if ($form != $this->pluginName) {
+            return false;
+        }
```
Comment (bembelimen) [inline plugins/schemaorg/event/src/Extension/Event.php:148]: Please add empty line

File: plugins/schemaorg/event/src/Extension/Event.php [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+        $this->storeSchemaToStandardLocation($event);
```
Comment (bembelimen) [inline plugins/schemaorg/event/src/Extension/Event.php:150]: Please add empty line

File: administrator/components/com_admin/sql/updates/mysql/4.2.3-2022-12-10.sql (hunk 1 above)
Comment (shazmasiddiqui) [reply administrator/components/com_admin/sql/updates/mysql/4.2.3-2022-12-10.sql]: done

**Pull request merged**